packages/statemanager/DEVELOPER.md

```diff
@@ -10,3 +10,35 @@ Tests can be found in the `tests` directory. The StateManager can be tested as:
 ### CI Test Integration
 
 Tests and checks are run in CI using [Github Actions](https://github.com/ethereumjs/ethereumjs-monorepo/actions). The configuration can be found in `.github/workflows`.
+
+## Debugging
+
+For the debugging output to show up, set the `DEBUG` environment variable, e.g. in Linux/Mac OS:
+`export DEBUG=ethjs,*`
+
+`ethjs` **must** be included in the `DEBUG` environment variables to enable **any** logs.
+Additional log selections can be added with a comma separated list (no spaces). Logs with extensions can be enabled with a colon `:`, and `*` can be used to include all extensions.
+
+`DEBUG=ethjs,thislog,thatlog,otherlog,otherlog:sublog,anotherLog:* node myscript.js`
+
+The following initial logger is currently available:
+
+| Logger                                     | Description                                              |
+| ------------------------------------------ | -------------------------------------------------------- |
+| `statemanager:MerkleStateManager`          | Operations happening on the `MerkleStateManager`         |
+| `statemanager:RPCStateManager`             | Operations happening on the `RPCStateManager`            |
+| `statemanager:StatefulVerkleStateManager`  | Operations happening on the `StatefulVerkleStateManager` |
+| `statemanager:StatelessVerkleStateManager` | Operations accessing verkle witnesses                    |
```

Comment:
Ah, it's generally a good idea to get consistent here! 👍 Can we nevertheless go the otherway around here and *shorten* all names, so `merkle`, `rpc` and for `verkle` it might even be ok to for both stay in the same namespace (since statemanagers aren't really used concurrently in most/all use cases, so no risk of mixing things up)?

In practice it just turned out that it becomes annying pretty quickly if logger names are too long, especially if one wants to concatenate 2,3 or 4 of them.

Also - the thing Jochem mentioned - debug loggers should always be lowercase (so: `statemanager` is also lowercase, just as an example) - I think this was also some discussion in the past and it generally makes sense since then its a lot easier to adhoc remember and not stumble over not casing things right on usage.

Reply:
I made all of the namespaces that we declare lowercase, but as mentioned just bellow, namespaces are case sensitive, and setting the debug to trace e.g. `trie:put` will not work while `trie:PUT` will, so there is going to be some inconsistency here.